RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractRecycleBin.java

```diff
@@ -0,0 +1,552 @@
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, String.format("%s may not have been created.", RECYCLE_BIN_ARTIFACT_NAME), ex);
+        }
+
+        BlackboardArtifact.Type recycleBinArtifactType;
+
+        try {
+            recycleBinArtifactType = tskCase.getArtifactType(RECYCLE_BIN_ARTIFACT_NAME);
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, String.format("Unable to retrive custom artifact type %s", RECYCLE_BIN_ARTIFACT_NAME), ex); // NON-NLS
+            // If this doesn't work bail.
+            return;
+        }
+        Map<String, String> userNameMap;
+        
+        try {
+            userNameMap = makeUserNameMap(dataSource);
+        } catch (TskCoreException ex) {
+           logger.log(Level.WARNING, "Unable to create OS Account user name map", ex);
+           // This is not the end of the world we will just continue without 
+           // user names
+           userNameMap = new HashMap<>();
+        }
+        
+        FileManager fileManager = Case.getCurrentCase().getServices().getFileManager();
+
+        Map<String, List<AbstractFile>> rFileMap;
+        try {
+            rFileMap = makeRFileMap(dataSource);
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, String.format("Unable to create $R file map for dataSource: %s", dataSource.getName()), ex);
+            return; // No $R files, no need to continue;
+        }
+
+        List<AbstractFile> iFiles;
+
+        try {
+            iFiles = fileManager.findFiles(dataSource, "$I%"); //NON-NLS            
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, "Unable to find recycle bin I files.", ex); //NON-NLS
+            return;  // No need to continue
+        }
+        
+        String tempRARecycleBinPath = RAImageIngestModule.getRATempPath(Case.getCurrentCase(), "recyclebin"); //NON-NLS
+
+        for (AbstractFile iFile : iFiles) {
+
+            if (context.dataSourceIngestIsCancelled()) {
+                return;
+            }
+            
+            processIFiles(dataSource, context, recycleBinArtifactType, iFile, userNameMap, rFileMap, tempRARecycleBinPath);
+        }
+
+        (new File(tempRARecycleBinPath)).delete();
+    }
+    
+    /**
+     * Process each individual iFile.
+     * 
+     * @param dataSource
+     * @param context
+     * @param recycleBinArtifactType Module created artifact type
+     * @param iFile The AbstractFile to process
+     * @param userNameMap Map of user ids to names
+     * @param tempRARecycleBinPath Temp directory path
+     */
+    private void processIFiles(Content dataSource, IngestJobContext context,  BlackboardArtifact.Type recycleBinArtifactType, AbstractFile iFile,  Map<String, String> userNameMap, Map<String, List<AbstractFile>> rFileMap, String tempRARecycleBinPath) {
+        String tempFilePath = tempRARecycleBinPath + File.separator + iFile.getName();
```

Comment:
We should make this more unique by adding an ID to the end of it. Could have naming collisions between volumes on the same data source.